### ISSUE TITLE
static files do not work on Heroku

### ISSUE THREAD

**Issue by morenoh149** [pydanny/cookiecutter-django#2354]
## What happened?
On heroku we get errors for static files like
```
Navigated to https://app.herokuapp.com/
app.herokuapp.com/:1 Refused to apply style from 'https://app.herokuapp.com/static/CACHE/css/output.5f8f70551f02.css' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled.
app.herokuapp.com/:1179 GET https://app.herokuapp.com/static/CACHE/js/output.83a0993870e8.js net::ERR_ABORTED 404 (Not Found)
```

## What should've happened instead?
Static files should simply work on heroku. Do whatever [django-heroku](https://devcenter.heroku.com/articles/django-app-configuration#settings-py-changes) does.

## Steps to reproduce

[//]: make a django-cookiecutter app
[//]: deploy to heroku
[//]: attempt to load a page that uses css or js

on heroku the env var `DJANGO_SETTINGS_MODULE` is `config.settings.production`.

Heroku outputs the following during build
```
$ gp heroku                                             
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 12 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 346 bytes | 346.00 KiB/s, done.                                              
Total 3 (delta 2), reused 0 (delta 0)
remote: Compressing source files... done.
remote: Building source:
remote:
remote: -----> Python app detected
remote:  !     Python has released a security update! Please consider upgrading to python-3.7.5
remote:        Learn More: https://devcenter.heroku.com/articles/python-runtimes
remote: -----> Installing SQLite3
remote: Sqlite3 successfully installed.
remote: -----> Installing requirements with pip
remote:
remote: -----> $ python manage.py collectstatic --noinput
remote:        158 static files copied to '/tmp/build_64a4468323052c69a0e5f3e6dbabba45/staticfiles', 470 post-processed.
remote:
remote: -----> Discovering process types
remote:        Procfile declares types -> release, web, worker
remote:
remote: -----> Compressing...
remote:        Done: 75.5M
remote: -----> Launching...
remote:  !     Release command declared: this new release will not be available until the command succeeds.
remote:        Released v23
remote:        https://app.herokuapp.com/ deployed to Heroku
remote:
remote: Verifying deploy... done.
remote: Running release command...
   b5644f6..7ef0bb1  master -> master
```
we see that django is detected and [heroku runs collectstatic automatically for django apps](https://devcenter.heroku.com/articles/django-assets#collectstatic-during-builds). However the destination is `/tmp/build_<number>/staticfiles`. The django cookiecutter app looks for static assets in `<hostname>/static/CACHE/{css,js}/<compressed_staticfile>.{js,css}`.

Maybe the production config needs to be updated to point to this `tmp` dir heroku makes.

**Comment by luzfcb** [comment]
What are the exact options you chose when generating the project?

**Comment by morenoh149** [comment]
I think I chose docker and heroku. If that is not possible I chose docker. And when I couldn't find instructions for running docker on heroku I used the Procfile provided to deploy to heroku.

**Comment by browniebroke** [comment]
I know Heroku has (had?) some support for Docker, however this combination (Heroku with Docker) isn't supported by Cookiecutter Django at the moment. So Heroku without Docker it should be.

Do you happen to remember what options you've selected related to static files? 

- Use Whitenoise?
- Cloud provider?
- Use compressor?
- JS task runner?
- Custom Bootstrap compilation?

**Comment by browniebroke** [comment]
Potentially fixed by #2442 although you'll probably hit this one now: #2581

**Comment by morenoh149** [comment]
No clue. Maybe you can have the cookiecutter remember the options selected and the date run on the dev machine?

We ended up writing our docker files and docker compose that would work for dev and heroku.
